### DISCUSSION TITLE
What is general guidance around analysis cache thrashing when invoking bazel build --> bazel coverage --> bazel build?

### DISCUSSION THREAD

**Discussion by Ryang20718** [bazelbuild/bazel#22655] [Q&A]
`--collect_code_coverage and --instrumentation_filter` causes the analysis cache to thrash in CI when running bazel build --> bazel coverage

What do most people do currently to avoid this?

separate output_base?

**Comment by brentleyjones** [comment]
We add those flags to `build` as well.

**Reply by Ryang20718** [reply]
@brentleyjones would those flags for build have any unwanted side effects?

**Reply by brentleyjones** [reply]
They would cause your builds to have instrumentation added to them. Mainly just bloat, but you get cache hits on them.

**Reply by keith** [reply]
probably just need to make sure you don't do that for your prod builds
